### PR TITLE
[FLINK-7037] Remove scala suffic from flink-examples module

### PR DESCRIPTION
**Pull request by zentol**
This PR removes the scala version suffix from the flink-examples module.

The suffix prevented building single modules that rely on any example module. This does not cause issues when building flink completely as we traverse the modules from the parent, allowing the scala version to be resolved for all modules.

When building a single module, in order to resolve the scala version we have to traverse from the target module towards the parent, which isn't possible since the scala version isn't resolved, i.e. we cant resolve the scala version since we can't find the parent, and we can't fine the parent since the scala version wasn't resolved.

### CODE REVIEW

File: flink-examples/flink-examples-table/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@ under the License.
 
 	<parent>
 		<groupId>org.apache.flink</groupId>
-		<artifactId>flink-examples_${scala.binary.version}</artifactId>
+		<artifactId>flink-examples}</artifactId>
```
Comment (greghogan) [inline flink-examples/flink-examples-table/pom.xml]: Leftover `}`.

**Comment by zentol** [comment]
@greghogan fixed

**Comment by greghogan** [comment]
+1. Test failure is unrelated.

**Comment by zentol** [comment]
merging.

**Comment by aljoscha** [comment]
@zentol why was this only a problem for the example module?

**Comment by zentol** [comment]
@aljoscha Because it is the only parent module that has a scala suffix. All other modules with a scala suffix don't have children.

**Comment by aljoscha** [comment]
Thanks! That makes sense.

**Pull request closed without merging**